src/app/modules/sgid/__tests__/sgid.controller.spec.ts

```diff
@@ -0,0 +1,181 @@
+import { err, errAsync, ok, okAsync } from 'neverthrow'
+import { mocked } from 'ts-jest/utils'
+
+import config from 'src/app/config/config'
+import * as RealFormService from 'src/app/modules/form/form.service'
+import { MOCK_COOKIE_AGE } from 'src/app/modules/myinfo/__tests__/myinfo.test.constants'
+
+import expressHandler from 'tests/unit/backend/helpers/jest-express'
+
+import { ApplicationError } from '../../core/core.errors'
+import { FormNotFoundError } from '../../form/form.errors'
+import * as SgidController from '../sgid.controller'
+import {
+  SgidFetchAccessTokenError,
+  SgidFetchUserInfoError,
+  SgidInvalidStateError,
+} from '../sgid.errors'
+import { sgidService as realSgidService } from '../sgid.service'
+
+import {
+  MOCK_AUTH_CODE,
+  MOCK_COOKIE_SETTINGS,
+  MOCK_DESTINATION,
+  MOCK_JWT,
+  MOCK_REMEMBER_ME,
+  MOCK_SGID_FORM,
+  MOCK_SP_FORM,
+  MOCK_STATE,
+  MOCK_TARGET,
+  MOCK_TOKEN_RESULT,
+  MOCK_USER_INFO,
+} from './sgid.test.constants'
+
+jest.mock('../sgid.service')
+const sgidService = mocked(realSgidService, true)
+jest.mock('src/app/modules/form/form.service')
+const FormService = mocked(RealFormService, true)
+jest.mock('src/app/config/config')
+const MockConfig = mocked(config, true)
+MockConfig.isDev = false
+
+const MOCK_RESPONSE = expressHandler.mockResponse()
+const MOCK_LOGIN_REQ = expressHandler.mockRequest({
+  query: { code: MOCK_AUTH_CODE, state: MOCK_STATE },
+})
+
+describe('sgid.controller', () => {
+  beforeEach(() => jest.clearAllMocks())
+  afterAll(() => jest.clearAllMocks())
+
+  describe('handleLogin', () => {
+    const { handleLogin } = SgidController
```

Comment:
nit: should we invoke the full `SgidController.handleLogin` instead of destructuring it here? Allows the reader to know where the function originates amongst all the other functions being called :P